src/components/Person.js

```diff
@@ -5,150 +5,136 @@ import store from 'store2';
-    // this callback would have the parent update its own min/max rows and pass this to all children
-    // but it seems like for this to work you need child.state.height (to figureo ut the new min rows)
+  updateSave = () => {
+    while (updated_niceties_spinlock) {}
+    let addString;
+    if (this.props.data.stints.length > 0) {
+        addString = this.props.data.id + "," + this.props.data.stints[this.props.data.stints.length - 1].end_date;
+    } else {
+        addString = this.props.data.id + ",2016-11-03";
+    }
+    if (!(addString in this.props.updated_niceties)) {
+        this.props.updated_niceties.add(addString);
+    }
+    store.set("saved", false);
+    this.props.saveReady();
+  }
 
-    render: function() {
-        let anonymousRender;
-        if (this.state.anonValue === true) {
-            anonymousRender = (
-                <Checkbox checked onChange={this.anonymousChange}>
-                    Submit Anonymously
-                </Checkbox>
-            );
-        } else if (this.state.anonValue === false) {
-            anonymousRender = (
-                <Checkbox onChange={this.anonymousChange}>
-                    Submit Anonymously
-                </Checkbox>
-            );
-        }
+  textareaChange = (event) => {
+    this.setState({textValue: event.target.value});
+    store.set("nicety-" + this.props.data.id, event.target.value);
+    this.updateSave();
+  }
 
-        let noReadRender;
-        if (this.state.noReadValue === true) {
-            noReadRender = (
-                <Checkbox checked onChange={this.noReadChange}>
-                    Don't Read At Ceremony
-                </Checkbox>
-            );
-        } else if (this.state.noReadValue === false) {
-            noReadRender = (
-                <Checkbox onChange={this.noReadChange}>
-                    Don't Read At Ceremony
-                </Checkbox>
-            );
-        }
-        return (
-            <div className="person">
-                <Image responsive={true} src={this.props.data.avatar_url} circle={true} />
-                <h3>{this.props.data.name}</h3>
-            <textarea
-                defaultValue={this.state.textValue}
-                onChange={this.textareaChange}
-                rows="6"
-                placeholder={this.props.data.placeholder}
-            />
-            {anonymousRender}
-            {noReadRender}
-            </div>
+  anonymousChange = (event) => {
+    this.setState({anonValue: event.target.checked});
+    store.set("anonymous-" + this.props.data.id, event.target.checked);
+    this.updateSave();
+  }
+
+  noReadChange = (event) => {
+    this.setState({noReadValue: event.target.checked});
+    store.set("no_read-" + this.props.data.id, event.target.checked);
+    this.updateSave();
+  }
+
+  render() {
+    let anonymousRender = (
+            <Checkbox checked={this.state.anonValue === true}
```

Comment:
Big +1 to this! We could go further, even, by inlining the variable - we don't need `anonymousRender` or `noReadRender` any more, and could just use the `<Checkbox>`es directly in the returned JSX.